mne/beamformer/tests/test_ap.py

```diff
@@ -0,0 +1,169 @@
+# Authors: Yuval Realpe <yuval.realpe@gmail.com>
+#
+# License: BSD-3-Clause
+
+import os.path as op
+
+import numpy as np
+from scipy import linalg
+from random import choice
+
+import mne
+from mne.beamformer import ap
+from mne.cov import regularize
+from mne.datasets import testing
+
+
+data_path = testing.data_path(download=False)
+fname_ave = op.join(data_path, 'MEG', 'sample', 'sample_audvis-ave.fif')
+fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis-cov.fif')
+fname_fwd = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-meg-eeg-oct-4-fwd.fif')
+
+
+def _get_data(ch_decim=1):
+    """Read in data used in tests."""
+    # Read evoked
+    condition = choice(['Left Auditory',
+                        'Right Auditory',
+                        'Left visual',
+                        'Right visual'])
+    evoked = mne.read_evokeds(fname_ave, condition=condition,
+                              baseline=(None, 0))
+    evoked.crop(tmin=0.05, tmax=0.15)
+    picks = mne.pick_types(evoked.info, meg=True, eeg=False)
+    picks = picks[::ch_decim]
+    evoked.pick_channels([evoked.ch_names[pick] for pick in picks])
+    evoked.info.normalize_proj()
+
+    # Read noise_cov
+    noise_cov = mne.read_cov(fname_cov)
+    noise_cov['projs'] = []
+    noise_cov = regularize(noise_cov, evoked.info, rank='full', proj=False)
+
+    return evoked, noise_cov
+
+
+def simu_data_2src(evoked, forward, noise_cov, times, nave=1):
```

Comment:
this function is copy pasted from test_rap_music.py

you can try to import it from this test file to avoid the copy/paste

Reply:
changed to import simu_data() from test_rap_music.py